include/EASTL/map.h

```diff
@@ -439,6 +442,47 @@ namespace eastl
 	#endif
 
 
+	template <typename Key, typename T, typename Compare, typename Allocator>
+	inline T& map<Key, T, Compare, Allocator>::at(const Key& key)
+	{
+		iterator itLower(lower_bound(key)); // itLower->first is >= key.
+
+		#if EASTL_EXCEPTIONS_ENABLED
+			if(itLower == end())
```

Comment:
Since the branch is checked in both cases, I'd promote it outside of the preprocessor guards.